models/classes/files/QtiFlysystemFile.php

```diff
@@ -48,7 +48,10 @@ public function getMimeType()
 
         // The parent function will return "text/plain" when the mime type can't be detected. As the last resort,
         // we use the original file name when available to try to detect its mime type.
-        if ($mimeType === 'text/plain' && $this->hasFilename()) {
+        if (
+            in_array($mimeType, ['text/plain', 'application/octet-stream'])
+            && $this->hasFilename()
```

Comment:
`hasFilename()` doesn't check if the file name stored is not empty. I do hope it's impossible to store a file with an empty name, though :)

Reply:
It feels much safer to do the following.

```
$mimeType = parent::getMimeType();
$fileName = $this→getFileName();

if ($fileName && in_array($mimeType, ['text/plain', 'application/octet-stream'])) {
    ...
}
```